config/config.go

```diff
@@ -22,7 +23,8 @@ func GetOneLoginProvider(p string) (*OneLoginProviderConfig, error) {
 	clientSecret := viper.GetString(fmt.Sprintf("providers.%s.client-secret", p))
 	clientID := viper.GetString(fmt.Sprintf("providers.%s.client-id", p))
 	subdomain := viper.GetString(fmt.Sprintf("providers.%s.subdomain", p))
-	user := viper.GetString(fmt.Sprintf("providers.%s.username", p))
+	username := viper.GetString(fmt.Sprintf("providers.%s.username", p))
+	region := viper.GetString(fmt.Sprintf("providers.%s.region", p))
```

Comment:
At the moment there is no verification for the `region` flag. On the other hand, we currently have hardcoded regions in the code. I would either verify that `region` is either `EU` or `US`, **OR** decide we allow any region value and simply template this value into the URL. I think the 1st option is probably safer. At the moment I'm getting a segfault when using the new flag. Checking why.

Reply:
In https://github.com/allcloud-io/clisso/blob/6aaee3f8aa90e2e8e63016f0af4e7519a2bab9c0/onelogin/endpoints.go#L25-L41 we verify whether the region is correct when using it, and in https://github.com/allcloud-io/clisso/blob/6aaee3f8aa90e2e8e63016f0af4e7519a2bab9c0/onelogin/endpoints.go#L25-L41 we check whether it's empty.

I suspect this case is where a region isn't set- do you get a similar segfault if, say, subdomain is empty?

Reply:
Right. Well, it still might be worth doing the validation as soon as possible. I indeed get the error above when configuring a wrong region. The segfault was due to a nil pointer in `NewClient`.

Reply:
See latest commit:

```bash
# No region set
$(onelogin-eu) ./clisso providers create onelogin foo --client-id a --client-secret a --subdomain b
Provider 'foo' saved to config file

# Incorrect region set
$(onelogin-eu) ./clisso providers create onelogin foo2 --client-id a --client-secret a --subdomain b --region NONE
Region must be one of US or EU
```